src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -511,15 +511,14 @@ public QueryResult<QueryAddressItem> AddressHistory(string address, int offset,
          // This will first perform one db query.
          long total = addressComputedTable.CountSent + addressComputedTable.CountReceived + addressComputedTable.CountStaked + addressComputedTable.CountMined;
 
-         if (offset == 0 || offset > total)
-            offset = (int)total;
+         // Filter by the position, in the order of first entry being 1 and then second entry being 2.
+         filter = filter.OrderBy(s => s.Position);
```

Comment:
Perhaps it should be a good idea to also change the position sort index?
Currently it's Descending
https://github.com/block-core/blockcore-indexer/blob/master/src/Blockcore.Indexer.Core/Sync/SyncTasks/BlockIndexer.cs#L241

Reply:
This is an issue no so we can fix it later